### PR TITLE
#343 ignore accents in search

### LINKED ISSUE #343: User and room search should ignore accents.
### Subject of the issue

When searching for a User in the sidebar, or a Room in the appbar, the search won't bring users if you don't include accents in the name.

### Steps to reproduce

1. Have a user or room with an accent somewhere in the name (eg: José/Café)
2. Search for their name without including an accent (jose)

Example:

matrix

### Expected behaviour

The filter function should ignore accents when searching for users.

### Actual behaviour

It doesn't, and those users won't show up.

### Possible Solution

Just make it so the filter ignores accents. 

For ES6, [`String.prototype.normalize` ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize) can be used, as recommended by [this stack overflow answer](https://stackoverflow.com/a/37511463/4802649).

### PR DESCRIPTION
**Pull request by phiter**
### Description
Resolves #343 

### How to test?
1. Have a user or room with an accent somewhere in the name (eg: José/Café)
2. Search for their name without including an accent (jose)

### Expected behavior
User should show up even if the search query doesn't include the accent.

#### Proof:

xxx

### CODE REVIEW

File: frontend/src/morpheus/store/reducers.js [hunk 1]
```diff
@@ -56,6 +56,8 @@ export const initialState = {
   }
 };
 
+const normalize = str => str.normalize("NFD").replace(/[\u0300-\u036f]/g, "").toLowerCase();
```
Comment (megatroom) [inline frontend/src/morpheus/store/reducers.js:59]: Nice job here! :clap: :clap: :clap:

Comment (phiter) [reply frontend/src/morpheus/store/reducers.js:59]: i'll make the requested changes and push in a sec

**Comment by phiter** [comment]
> Excellent improvement! Direct, clean and very valuable 🥇
> 
> Do you mind creating some unit tests in [reducers.test.js](https://github.com/ResultadosDigitais/matrix/blob/master/frontend/test/morpheus/store/reducers.test.js) to address those cases described in the PR description?

Done!

**Comment by phiter** [comment]
Codeclimate hates me lol

**Comment by megatroom** [comment]
> Codeclimate hates me lol

ahahhahahahah

I thought about making the same suggestion in my review, but found it meaningless given the change. But it seems that Codeclimate is more rigorous than me  :rofl:

**Pull request merged**